app/models/library.rb

```diff
@@ -165,6 +164,18 @@ def file_last_modified(file)
     end
   end
 
+  def create_path_if_not_on_disk=(val)
+    if val == "1" && storage_service == "filesystem"
+      begin
+        FileUtils.makedirs(path)
+      rescue Errno::EROFS
+        errors.add(:path, "must be writable")
+      rescue Errno::EACCES
```

Comment:
you can rescue both errors in one line, I think, if you separate them with a comma: `rescue Errno::EROFS, Errno::EACCES`